### ISSUE TITLE
https://akispetretzikis.com stopped working 

### ISSUE THREAD

**Issue by giannis371** [hhursev/recipe-scrapers#1235]
3 months ago I added a recipe to tandoor app but now it's not working anymore. Other websites work.
A recipe link 
https://akispetretzikis.com/en/recipe/1564/pagwto-fistiki

**Comment by jayaddison** [comment]
Hi @giannis371 - thanks for the bugreport!  Could you confirm the version of `recipe-scrapers` in use in your Tandoor instance, and, if possible, any error output that occurs when attempting to read a recipe from that URL?

**Comment by giannis371** [comment]
Hi @jayaddison , thanks for the fast reply.
Version is
recipe-scrapers==15.0.0
Error I get is
Failure there was an error importing the recipe!
@smilerz from tandoor found out that the page is a loading page without any recipe data.

**Comment by smilerz** [comment]
It appears the recipe is loaded asynchronously.

From Tandoor:
   ` scrape = scrape_html(org_url=url, html=html, supported_only=False)`

```
  File "/opt/recipes/venv/lib/python3.12/site-packages/recipe_scrapers/__init__.py", line 844, in scrape_html

    return SCRAPERS[host_name](html=html, url=org_url)

  File "/opt/recipes/venv/lib/python3.12/site-packages/recipe_scrapers/akispetretzikis.py", line 10, in __init__

    self.soup.find("script", {"id": "__NEXT_DATA__"}).get_text()

AttributeError: 'NoneType' object has no attribute 'get_text'
```

**Comment by jayaddison** [comment]
Thanks @giannis371 @smilerz.

I was able to scrape the same recipe URL successfully using a standalone checkout of `recipe-scrapers` v15.0.0 (without Tandoor).

Note: `recipe-scrapers` doesn't contain any asynchronous request code, as far as I'm aware.

**Comment by jayaddison** [comment]
However.. I did not set `supported_only=False` during my repro attempt.  I will inspect the code and/or attempt again related to that soon.

**Comment by smilerz** [comment]
> However.. I did not set `supported_only=False` during my repro attempt. I will inspect the code and/or attempt again related to that soon.

weird - I get the same error running both  
`scrape = scrape_html(org_url=url, html=html, supported_only=False) `
and
`scrape = scrape_html(org_url=url, html=html)`

What are you using to collect the html?  Tandoor is using:
```
html = requests.get(
                                url,
                                headers={"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0"}
                            ).content
```

**Comment by jayaddison** [comment]
> What are you using to collect the html? Tandoor is using:
> [ ... snip ... ]

I was using the built-in HTTP headers from `recipe-scrapers` v15.0.0 (`online=True`) -- and so I think this is _probably_ a case where we have a site that produces different results based on the HTTP client / browser in use.

Note / reminder that more-recent versions of `recipe-scrapers` now declare a custom user-agent header (ref #1219) -- I have not tested that one yet, but will do also to compare.

**Comment by smilerz** [comment]
Now I'm really confused because `scrape_html(org_url=url, html=None, online=True)` generates the exact same error.

![image](https://github.com/user-attachments/assets/2fc19d20-c501-4436-adfc-6594b26a53ee)

**Comment by jayaddison** [comment]
I'm not sure what to suggest here; I'm able to access the recipe using both 15.0.0 and 15.1.0 -- so to me it would seem that there is at least one factor outside of this codebase that determines whether the recipe HTML is provided at request-time.

**Comment by giannis371** [comment]
Indeed that's very confusing.

**Comment by smilerz** [comment]
> I'm not sure what to suggest here; I'm able to access the recipe using both 15.0.0 and 15.1.0 -- so to me it would seem that there is at least one factor outside of this codebase that determines whether the recipe HTML is provided at request-time.

Agreed - I don't think this is software related.

**Comment by smilerz** [comment]
FWIW, once loaded the recipe metadata is still in Greek.  so even if it worked, not sure you would get what you were looking for.

**Comment by giannis371** [comment]
No you just have to change the language in English

smilerz ***@***.***> schrieb am Do., 5. Sept. 2024, 15:23:

**Comment by giannis371** [comment]
> FWIW, once loaded the recipe metadata is still in Greek. so even if it worked, not sure you would get what you were looking for.

No you can just change the language. Like a said 5 months ago I added a recipe from this website and it's still in my tandoor working correctly.

**Comment by giannis371** [comment]
> > So it means we just have to close the issue and remove the website from the scrappers-list or is there any chance to find a solution?
> 
> We don't have to close the issue, but at the moment it seems unclear whether there is a technical solution. What benefits would removing the recipe scraper provide?

Yes you are right. You said before it's working fine with your system so the problem is not not the scrapper. That means other people can still use it.

**Comment by smilerz** [comment]
> No you can just change the language. Like I said 5 months ago I added a recipe from this website and it's still in my tandoor working correctly.

They've obviously made a change how the website works.  I loaded the recipe using the bookmarklet and it's in Greek.
